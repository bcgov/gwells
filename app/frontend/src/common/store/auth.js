export const SET_KEYCLOAK = 'SET_KEYCLOAK'

const auth = {
  state: {
    keycloak: {}
  },
  mutations: {
    [SET_KEYCLOAK] (state, payload) {
      state.keycloak = payload
    }
  },
  getters: {
    keycloak (state) {
      return state.keycloak
    },
    userRoles (state) {
      if (state.keycloak && state.keycloak.authenticated) {
        return {
          registry: {
            // map SSO roles to web app permissions
            view: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority') ||
              state.keycloak.hasRealmRole('registries_viewer') ||
              state.keycloak.hasRealmRole('registries_adjudicator')),
            edit: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority') ||
              state.keycloak.hasRealmRole('registries_adjudicator')),
            approve: (state.keycloak.hasRealmRole('gwells_admin') ||
              state.keycloak.hasRealmRole('registries_statutory_authority'))
          },
          wells: {
            view: (!!state.keycloak.hasRealmRole('gwells_admin')),
            edit: (!!state.keycloak.hasRealmRole('gwells_admin')),
            approve: (!!state.keycloak.hasRealmRole('gwells_admin'))
          },
          submissions: {
            view: (!!state.keycloak.hasRealmRole('gwells_admin')),
            edit: (!!state.keycloak.hasRealmRole('gwells_admin')),
            approve: (!!state.keycloak.hasRealmRole('gwells_admin'))
          }
        }
      } else {
        return {
          registry: {},
          wells: {},
          submissions: {}
        }
      }
    }
  }
}

export default auth
