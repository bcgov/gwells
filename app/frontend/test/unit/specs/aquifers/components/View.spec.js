/*
    Licensed under the Apache License, Version 2.0 (the "License");
    you may not use this file except in compliance with the License.
    You may obtain a copy of the License at

        http://www.apache.org/licenses/LICENSE-2.0

    Unless required by applicable law or agreed to in writing, software
    distributed under the License is distributed on an "AS IS" BASIS,
    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
    See the License for the specific language governing permissions and
    limitations under the License.
*/

import { mount, createLocalVue } from '@vue/test-utils'
import ViewComponent from '@/aquifers/components/View.vue'
import Vuex from 'vuex'
import axios from 'axios'
import auth from '@/common/store/auth.js'
import aquiferCodes from '@/aquifers/store/codes'
import VueRouter from 'vue-router'

jest.mock('axios')

const localVue = createLocalVue()
localVue.use(Vuex)
localVue.use(VueRouter)

const aquiferFixture = {
  aquifer_id: '4',
  aquifer_name: 'Aggasiz',
  location_description: 'Aggasiz/Sea Bird Island',
  material_description: 'Sand and Gravel',
  litho_stratographic_unit: 'Fraser River Sediments',
  subtype_description: 'Unconfined sand and gravel - large river system',
  vulnerability_description: 'High',
  area: '63.3',
  productivity_description: 'High',
  demand: 'Low',
  mapping_year: '1993'
}

describe('View Component', () => {
  const component = (options) => mount(ViewComponent, {
    localVue,
    router: new VueRouter(),
    store: new Vuex.Store({
      modules: { auth, aquiferCodes }
    }),
    stubs: ['aquifer-documents'],
    ...options
  })

  it('queries aquifer on load', () => {
    const fetch = jest.fn()

    component({
      methods: { fetch }
    })

    expect(fetch).toHaveBeenCalled()
  })

  describe('View mode', () => {
    it('matches the snapshot', () => {
      const wrapper = component({
        data () {
          return {
            record: aquiferFixture
          }
        },
        methods: { fetch () {} },
        propsData: { edit: true }
      })

      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Edit mode', () => {
    it('matches the snapshot', () => {
      const wrapper = component({
        data () {
          return {
            record: aquiferFixture
          }
        },
        methods: { fetch () {} },
        propsData: { edit: true }
      })

      expect(wrapper.element).toMatchSnapshot()
    })

    describe('On save', () => {
      it('resets showSaveSuccess to false', () => {
        const wrapper = component({
          data () {
            return {
              record: aquiferFixture,
              showSaveSuccess: true
            }
          },
          methods: {
            fetch () {},
            navigateToView () {}
          },
          propsData: { edit: true }
        })

        axios.patch.mockResolvedValue(true)

        expect(wrapper.vm.showSaveSuccess).toBe(true)
        wrapper.vm.save()
        expect(wrapper.vm.showSaveSuccess).toBe(false)
      })

      it('resets fieldErrors to empty object', () => {
        const wrapper = component({
          data () {
            return {
              record: aquiferFixture,
              fieldErrors: { a: ['1'] }
            }
          },
          methods: {
            fetch () {},
            navigateToView () {}
          },
          propsData: { edit: true }
        })

        axios.patch.mockResolvedValue(true)

        wrapper.vm.save()
        expect(wrapper.vm.fieldErrors).toEqual({})
      })

      it('sends a patch with the contents of record on save', () => {
        const wrapper = component({
          methods: {
            fetch () {},
            navigateToView () {}
          },
          computed: { id () { return 10 } },
          data () { return { record: aquiferFixture } }
        })

        axios.patch.mockResolvedValue(true)

        wrapper.vm.save()

        expect(axios.patch).toHaveBeenCalledWith('aquifers/10/', aquiferFixture)
      })
    })
  })

  it('displays field errors messages', () => {
    let errorMessage = 'error message'

    const wrapper = component({
      data () {
        return {
          fieldErrors: {
            mapping_year: [errorMessage],
            aquifer_name: [errorMessage],
            litho_stratographic_unit: [errorMessage],
            location_description: [errorMessage],
            vulnerability: [errorMessage],
            material: [errorMessage],
            subtype: [errorMessage],
            quality_concern: [errorMessage],
            productivity: [errorMessage],
            area: [errorMessage],
            demand: [errorMessage],
            known_water_use: [errorMessage],
            notes: [errorMessage]
          }
        }
      },
      methods: { fetch () {} },
      propsData: { edit: true }
    })

    expect(wrapper.find('#aquifer-mapping-year + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-name + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-litho-stratigraphic-unit + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-location-description + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-vulnerability + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-material + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-subtype + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-quality-concern + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-productivity + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-area + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-demand + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-known-water-use + [role="alert"]').text()).toBe(errorMessage)
    expect(wrapper.find('#aquifer-notes + [role="alert"]').text()).toBe(errorMessage)
  })
})
