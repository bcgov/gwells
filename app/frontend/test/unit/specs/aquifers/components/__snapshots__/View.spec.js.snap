// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View Component Edit mode matches the snapshot 1`] = `
<div
  class="card p-3 mb-4"
>
  <!---->
   
  <!---->
   
  <div
    class="container"
  >
    <!---->
     
    <div
      class="row border-bottom mb-3 pb-2"
    >
      <div
        class="col"
      >
        <h4>
          Aquifer 4 Summary - Edit
        </h4>
      </div>
    </div>
     
    <!---->
     
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <div
            class="row"
          >
            <div
              class="aquifer-id col-form-label col-md-4"
            >
              
          Aquifer number
        
            </div>
             
            <div
              class="aquifer-id col-md-8"
            >
              
          4
        
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__49__BV_label_"
            class="b-form-group form-group"
            id="__BVID__49"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="aquifer_name"
                id="__BVID__49__BV_label_"
              >
                Aquifer name
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-name"
                  tabindex="1"
                  type="text"
                  value="Aggasiz"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__51__BV_label_"
            class="b-form-group form-group"
            id="__BVID__51"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="location_description"
                id="__BVID__51__BV_label_"
              >
                Descriptive location
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-location-description"
                  tabindex="3"
                  type="text"
                  value="Aggasiz/Sea Bird Island"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__53__BV_label_"
            class="b-form-group form-group"
            id="__BVID__53"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="material"
                id="__BVID__53__BV_label_"
              >
                Material type
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-material"
                  tabindex="5"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__55__BV_label_"
            class="b-form-group form-group"
            id="__BVID__55"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="quality_concern"
                id="__BVID__55__BV_label_"
              >
                Quality concerns
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-quality-concern"
                  tabindex="7"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__57__BV_label_"
            class="b-form-group form-group"
            id="__BVID__57"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="area"
                id="__BVID__57__BV_label_"
              >
                Size (kmÂ²)
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-area"
                  tabindex="9"
                  type="text"
                  value="63.3"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__59__BV_label_"
            class="b-form-group form-group"
            id="__BVID__59"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="known_water_use"
                id="__BVID__59__BV_label_"
              >
                Type of known water use
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-known-water-use"
                  tabindex="11"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__61__BV_label_"
            class="b-form-group form-group"
            id="__BVID__61"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="notes"
                id="__BVID__61__BV_label_"
              >
                Comments
              </label>
              <div
                class="col-sm-8"
              >
                <textarea
                  class="form-control"
                  id="aquifer-notes"
                  rows="4"
                  tabindex="13"
                  wrap="soft"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col-md-6"
        >
          <div
            aria-labelledby="__BVID__63__BV_label_"
            class="b-form-group form-group"
            id="__BVID__63"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="mapping_year"
                id="__BVID__63__BV_label_"
              >
                Year of mapping
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-mapping-year"
                  tabindex="2"
                  type="text"
                  value="1993"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__65__BV_label_"
            class="b-form-group form-group"
            id="__BVID__65"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="litho_stratographic_unit"
                id="__BVID__65__BV_label_"
              >
                Litho stratigraphic unit
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-litho-stratigraphic-unit"
                  tabindex="4"
                  type="text"
                  value="Fraser River Sediments"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__67__BV_label_"
            class="b-form-group form-group"
            id="__BVID__67"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="vulnerability"
                id="__BVID__67__BV_label_"
              >
                Vulnerability
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-vulnerability"
                  tabindex="6"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__69__BV_label_"
            class="b-form-group form-group"
            id="__BVID__69"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="subtype"
                id="__BVID__69__BV_label_"
              >
                Subtype
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-subtype"
                  tabindex="8"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__71__BV_label_"
            class="b-form-group form-group"
            id="__BVID__71"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="productivity"
                id="__BVID__71__BV_label_"
              >
                Productivity
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-productivity"
                  tabindex="10"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__73__BV_label_"
            class="b-form-group form-group"
            id="__BVID__73"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="demand"
                id="__BVID__73__BV_label_"
              >
                Demand
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-demand"
                  tabindex="12"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="row mt-4"
      >
        <div
          class="col-auto"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            
        Save
      
          </button>
           
          <button
            class="btn btn-default"
            type="button"
          >
            
        Cancel
      
          </button>
        </div>
      </div>
       
      <div>
        <div
          id="confirmSave___BV_modal_outer_"
        >
          <div
            aria-hidden="true"
            class="modal fade"
            id="confirmSave"
            role="dialog"
            style="display: none;"
          >
            <div
              class="modal-dialog modal-md"
            >
              <div
                aria-describedby="confirmSave___BV_modal_body_"
                aria-labelledby="confirmSave___BV_modal_header_"
                class="modal-content"
                role="document"
                tabindex="-1"
              >
                <header
                  class="modal-header"
                  id="confirmSave___BV_modal_header_"
                >
                  <h5
                    class="modal-title"
                  />
                  <button
                    aria-label="Close"
                    class="close"
                    type="button"
                  >
                    Ã
                  </button>
                </header>
                <div
                  class="modal-body"
                  id="confirmSave___BV_modal_body_"
                >
                  <p>
                    Are you sure you would like to save this record?
                  </p>
                </div>
                <footer
                  class="modal-footer"
                  id="confirmSave___BV_modal_footer_"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
       
      <div>
        <div
          id="confirmCancel___BV_modal_outer_"
        >
          <div
            aria-hidden="true"
            class="modal fade"
            id="confirmCancel"
            role="dialog"
            style="display: none;"
          >
            <div
              class="modal-dialog modal-md"
            >
              <div
                aria-describedby="confirmCancel___BV_modal_body_"
                aria-labelledby="confirmCancel___BV_modal_header_"
                class="modal-content"
                role="document"
                tabindex="-1"
              >
                <header
                  class="modal-header"
                  id="confirmCancel___BV_modal_header_"
                >
                  <h5
                    class="modal-title"
                  />
                  <button
                    aria-label="Close"
                    class="close"
                    type="button"
                  >
                    Ã
                  </button>
                </header>
                <div
                  class="modal-body"
                  id="confirmCancel___BV_modal_body_"
                >
                  <p>
                    Are you sure you want to quit editing this record?
                  </p>
                </div>
                <footer
                  class="modal-footer"
                  id="confirmCancel___BV_modal_footer_"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
     
    <!---->
     
    <h5
      class="mt-3 border-bottom"
    >
      Documentation
    </h5>
     
    <aquifer-documents-stub />
  </div>
</div>
`;

exports[`View Component View mode matches the snapshot 1`] = `
<div
  class="card p-3 mb-4"
>
  <!---->
   
  <!---->
   
  <div
    class="container"
  >
    <!---->
     
    <div
      class="row border-bottom mb-3 pb-2"
    >
      <div
        class="col"
      >
        <h4>
          Aquifer 4 Summary - Edit
        </h4>
      </div>
    </div>
     
    <!---->
     
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-6"
        >
          <div
            class="row"
          >
            <div
              class="aquifer-id col-form-label col-md-4"
            >
              
          Aquifer number
        
            </div>
             
            <div
              class="aquifer-id col-md-8"
            >
              
          4
        
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__12__BV_label_"
            class="b-form-group form-group"
            id="__BVID__12"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="aquifer_name"
                id="__BVID__12__BV_label_"
              >
                Aquifer name
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-name"
                  tabindex="1"
                  type="text"
                  value="Aggasiz"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__14__BV_label_"
            class="b-form-group form-group"
            id="__BVID__14"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="location_description"
                id="__BVID__14__BV_label_"
              >
                Descriptive location
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-location-description"
                  tabindex="3"
                  type="text"
                  value="Aggasiz/Sea Bird Island"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__16__BV_label_"
            class="b-form-group form-group"
            id="__BVID__16"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="material"
                id="__BVID__16__BV_label_"
              >
                Material type
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-material"
                  tabindex="5"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__18__BV_label_"
            class="b-form-group form-group"
            id="__BVID__18"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="quality_concern"
                id="__BVID__18__BV_label_"
              >
                Quality concerns
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-quality-concern"
                  tabindex="7"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__20__BV_label_"
            class="b-form-group form-group"
            id="__BVID__20"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="area"
                id="__BVID__20__BV_label_"
              >
                Size (kmÂ²)
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-area"
                  tabindex="9"
                  type="text"
                  value="63.3"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__22__BV_label_"
            class="b-form-group form-group"
            id="__BVID__22"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="known_water_use"
                id="__BVID__22__BV_label_"
              >
                Type of known water use
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-known-water-use"
                  tabindex="11"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__24__BV_label_"
            class="b-form-group form-group"
            id="__BVID__24"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="notes"
                id="__BVID__24__BV_label_"
              >
                Comments
              </label>
              <div
                class="col-sm-8"
              >
                <textarea
                  class="form-control"
                  id="aquifer-notes"
                  rows="4"
                  tabindex="13"
                  wrap="soft"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
         
        <div
          class="col-md-6"
        >
          <div
            aria-labelledby="__BVID__26__BV_label_"
            class="b-form-group form-group"
            id="__BVID__26"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="mapping_year"
                id="__BVID__26__BV_label_"
              >
                Year of mapping
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-mapping-year"
                  tabindex="2"
                  type="text"
                  value="1993"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__28__BV_label_"
            class="b-form-group form-group"
            id="__BVID__28"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="litho_stratographic_unit"
                id="__BVID__28__BV_label_"
              >
                Litho stratigraphic unit
              </label>
              <div
                class="col-sm-8"
              >
                <input
                  class="form-control"
                  id="aquifer-litho-stratigraphic-unit"
                  tabindex="4"
                  type="text"
                  value="Fraser River Sediments"
                />
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__30__BV_label_"
            class="b-form-group form-group"
            id="__BVID__30"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="vulnerability"
                id="__BVID__30__BV_label_"
              >
                Vulnerability
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-vulnerability"
                  tabindex="6"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__32__BV_label_"
            class="b-form-group form-group"
            id="__BVID__32"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="subtype"
                id="__BVID__32__BV_label_"
              >
                Subtype
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-subtype"
                  tabindex="8"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__34__BV_label_"
            class="b-form-group form-group"
            id="__BVID__34"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="productivity"
                id="__BVID__34__BV_label_"
              >
                Productivity
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-productivity"
                  tabindex="10"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
           
          <div
            aria-labelledby="__BVID__36__BV_label_"
            class="b-form-group form-group"
            id="__BVID__36"
            role="group"
          >
            <div
              class="form-row"
            >
              <label
                class="col-sm-4 col-form-label"
                for="demand"
                id="__BVID__36__BV_label_"
              >
                Demand
              </label>
              <div
                class="col-sm-8"
              >
                <select
                  class="form-control custom-select"
                  id="aquifer-demand"
                  tabindex="12"
                >
                  <option
                    value=""
                  />
                </select>
                <!---->
                <!---->
                <!---->
              </div>
            </div>
          </div>
        </div>
      </div>
       
      <div
        class="row mt-4"
      >
        <div
          class="col-auto"
        >
          <button
            class="btn btn-primary"
            type="button"
          >
            
        Save
      
          </button>
           
          <button
            class="btn btn-default"
            type="button"
          >
            
        Cancel
      
          </button>
        </div>
      </div>
       
      <div>
        <div
          id="confirmSave___BV_modal_outer_"
        >
          <div
            aria-hidden="true"
            class="modal fade"
            id="confirmSave"
            role="dialog"
            style="display: none;"
          >
            <div
              class="modal-dialog modal-md"
            >
              <div
                aria-describedby="confirmSave___BV_modal_body_"
                aria-labelledby="confirmSave___BV_modal_header_"
                class="modal-content"
                role="document"
                tabindex="-1"
              >
                <header
                  class="modal-header"
                  id="confirmSave___BV_modal_header_"
                >
                  <h5
                    class="modal-title"
                  />
                  <button
                    aria-label="Close"
                    class="close"
                    type="button"
                  >
                    Ã
                  </button>
                </header>
                <div
                  class="modal-body"
                  id="confirmSave___BV_modal_body_"
                >
                  <p>
                    Are you sure you would like to save this record?
                  </p>
                </div>
                <footer
                  class="modal-footer"
                  id="confirmSave___BV_modal_footer_"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
       
      <div>
        <div
          id="confirmCancel___BV_modal_outer_"
        >
          <div
            aria-hidden="true"
            class="modal fade"
            id="confirmCancel"
            role="dialog"
            style="display: none;"
          >
            <div
              class="modal-dialog modal-md"
            >
              <div
                aria-describedby="confirmCancel___BV_modal_body_"
                aria-labelledby="confirmCancel___BV_modal_header_"
                class="modal-content"
                role="document"
                tabindex="-1"
              >
                <header
                  class="modal-header"
                  id="confirmCancel___BV_modal_header_"
                >
                  <h5
                    class="modal-title"
                  />
                  <button
                    aria-label="Close"
                    class="close"
                    type="button"
                  >
                    Ã
                  </button>
                </header>
                <div
                  class="modal-body"
                  id="confirmCancel___BV_modal_body_"
                >
                  <p>
                    Are you sure you want to quit editing this record?
                  </p>
                </div>
                <footer
                  class="modal-footer"
                  id="confirmCancel___BV_modal_footer_"
                >
                  <button
                    class="btn btn-default"
                    type="button"
                  >
                    Cancel
                  </button>
                  <button
                    class="btn btn-primary"
                    type="button"
                  >
                    OK
                  </button>
                </footer>
              </div>
            </div>
          </div>
          <!---->
        </div>
      </div>
    </div>
     
    <!---->
     
    <h5
      class="mt-3 border-bottom"
    >
      Documentation
    </h5>
     
    <aquifer-documents-stub />
  </div>
</div>
`;
